src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs

```diff
@@ -362,6 +373,14 @@ internal static void ApplySchemaReferenceId(this JsonNode schema, JsonSchemaExpo
         {
             schema[OpenApiConstants.SchemaId] = schemaReferenceId;
         }
+        // If the type is a non-abstract base class that is not one of the derived types then mark it as a base schema.
+        if (context.TypeInfo.PolymorphismOptions is { } polymorphismOptions &&
+            context.BaseTypeInfo == context.TypeInfo &&
```

Comment:
Could the code for this check be shared?  It seems like it would be easy for the two implementations to get slightly out of sync.

Reply:
It can be, it's admittedly difficult to come up with a good method name for the shared logic but I think I did decent enough.